cmd/internal/cli/build_linux.go

```diff
@@ -510,7 +510,13 @@ func processDefs(args []string, argFile string, defs []types.Definition) ([]type
 	diff := provideKeys.Difference(matchedKeys)
 	if !diff.Empty() {
 		vars := strings.Fields(strings.TrimPrefix(diff.String(), "HashSet"))
-		return defs, fmt.Errorf("while matching all variables, there are unmatched %s variables", strings.Join(vars, ""))
+
+		// All mismatched keys are in provided keys
+		if buildArgs.buildArgsUnusedWarn {
+			sylog.Warningf("While matching all variables, there are unmatched %s variables", strings.Join(vars, ""))
```

Comment:
Wouldn't that print all the variables without a space between them?

I suggest instead:
```
sylog.Warningf("Unused build args: %s", strings.Join(vars, " "))
```

Reply:
thanks! resolved here
https://github.com/apptainer/apptainer/pull/1462/commits/93029198be71cad69e4c5391cf6514ccbebae11f